project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/c4m_user_notifications.module

```diff
@@ -293,18 +318,45 @@ function c4m_user_notifications_form_og_ui_add_users_alter(&$form, &$form_state)
 }
 
 /**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Triggers email notification when group administrator rejects membership
+ * request.
+ */
+function c4m_user_notifications_form_c4m_og_reject_membership_request_alter(&$form, &$form_state) {
+  $form['#submit'][] = 'c4m_user_notifications_moderated_group_notify_user_membership_rejected';
+  $form['#submit'][] = 'c4m_og_route_to_group_people';
+}
+
+/**
  * Implements hook_flag_flag().
  *
  * Triggers email notification when recommendation flag is set.
  */
 function c4m_user_notifications_flag_flag($flag, $entity_id, $account, $flagging) {
-  if ($flag->name != 'subscribe_c4m_follow_node') {
+  // Don't send notifications, if activated from CLI (dummy content migration).
+  if (drupal_is_cli()) {
+    return;
+  }
+
+  if (user_is_anonymous()) {
+    // If anonymous user, return early - no notifications.
+    return;
+  }
+
+  if ($flag->name != 'c4m_og_content_recommend') {
     // If not recommendation flag, return early.
     return;
   }
 
-  c4m_user_notifications_notify_author_content_recommended($flag, $entity_id, $account, $flagging);
-  c4m_user_notifications_notify_subscribed_editor_content_recommended($flag, $entity_id, $account, $flagging);
+  $node = node_load($entity_id);
+  $notification_types = array('wiki_page', 'discussion', 'document', 'event');
```

Comment:
Ok thanks.
Do we need to check this on multiple places, or is this the only place? Otherwise it might be worth abstracting this so this list can be managed in 1 place?